sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala

```diff
@@ -95,9 +95,128 @@ object CodeFormatter {
     new CodeAndComment(code.result().trim(), map)
   }
 
-  def stripExtraNewLinesAndComments(input: String): String = {
+  def stripExtraNewLinesAndCommentsUsingRegexp(input: String): String = {
     extraNewLinesRegexp.replaceAllIn(commentRegexp.replaceAllIn(input, ""), "\n")
   }
+
+  private object State extends Enumeration {
+    val TEXT, SEPARATOR, SEPARATOR_WITH_NEWLINE, MAYBE_COMMENT, LINE_COMMENT, BLOCK_COMMENT,
+    BLOCK_COMMENT_WITH_NEWLINE, MAYBE_BLOCK_COMMENT_CLOSING,
+    MAYBE_BLOCK_COMMENT_WITH_NEWLINE_CLOSING = Value
+  }
+
```

Comment:
Could you leave some comments here about why we need to replace the existing regex with this the custom parser?

Reply:
ok, done